design/object-bucket/object-bucket.md

```diff
@@ -0,0 +1,190 @@
+  ownerReferences: [3]
+  - name: SOME-CEPH-CLUSTER
+    kind: CephCluster
+    blockOwnerDeletion: true
+    <TRUCANTED FOR CLARITY>
+  labels:
+    ceph.rook.io/object/claims: [4]
+spec:
+  storageClassName: SOME-OBJECT-STORE [5]
+  bucketPrefix: PREFIX [6]
+status:
+  phase: {"pending", "available", "error"}  [7]
+  configMapRef: objectReference{}  [8]
+```
+
+1. `name` of the CephObjectBucketClaim composes part of the ConfigMap name
+1. `namespace` of the CephObjectBucketClaim, determines the namespace of the ConfigMap
+1. `ownerReference` sets the CephObjectBucketClaim as a child of the Ceph Cluster.  Cluster deletion causes the deletion of the ConfigMap 
+1. Label `ceph.rook.io/object/claims` associates all artifacts under the CephObjectBucketClaim operator. Set by operator.
+1. `storageClassName` is used to target the desired Object Store.  Used by the operator to get the Object Store service URL.
+1. `bucketPrefix` the operator will utilize the k8s.io/apiserver `names` package for generating random 5 character suffixes
+1. `phase` 3 possible phases of bucket creation, mutually exclusive
+1. `configMapRef` is an objectReference to the generated ConfigMap 
+
+### ConfigMap
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: rook-ceph-object-bucket-MY-BUCKET-1 [1]
+  namespace: USER-NAMESPACE [2]
+  labels:
+    ceph.rook.io/object/claims: [3]
+  ownerReferences: [4]
+  - name: MY-BUCKET-1
+    kind: CephObjectBucketClaim
+    blockOwnerDeletion: true
+    <TRUCANTED FOR CLARITY>
+data: 
+  CEPH_BUCKET_HOST: http://MY-STORE-URL [5]
+  CEPH_BUCKET_PORT: 80 [7]
+  CEPH_BUCKET_NAME: MY-BUCKET-1 [6]
+  CEPH_BUCKET_SSL: no [8]
+```
+1. `name` composed from `rook-ceph-object-bucket-` and CephObjectBucketClaim `metadata.name` value concatenated
+1. `namespace` determined by the namespace of the CephObjectBucketClaim
+1. Label `ceph.rook.io/object/claims` associates all artifacts under the CephObjectBucketClaim operator. Set by operator.
+1. `ownerReference` sets the ConfigMap as a child of the CephObjectBucketClaim. Deletion of the CephObjectBucketClaim causes the deletion of the ConfigMap
+1. `CEPH_BUCKET_HOST` host URL
+1. `CEPH_BUCKET_PORT` host port
+1. `CEPH_BUCKET_NAME` bucket name
+1. `CEPH_BUCKET_SSL` boolean representing SSL connection
+
+### StorageClass
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: SOME-OBJECT-STORE
+  labels: 
+    ceph.rook.io/object/claims: [1]
+provisioner: "" [2]
+parameters:
+  objectStore: MY-STORE [3]
+  objectStoreNamespace: MY-STORE-NAMESPACE [4]
+```
+
+1. Label `ceph.rook.io/object/claims` associates all artifacts under the CephObjectBucketClaim operator.  Defined in example StorageClass and set by cluster admin.  
+1. `provisioner` value ignored, only the Rook-Ceph operator will watch CephObjectBucketClaims
+1. `objectStore` used by the operator to derive the object store Service name.
+1. `objectStoreNamespace` the namespace of the object store
```

Comment:
What do you think about showing the Owner and User secret here too?

Reply:
> What do you think about showing the Owner ...

this is a function of the CephObjectStoreUser controller and not an implementation of this proposal

> The User secret would probably have its OwnerReference set to the COBC.

This depends on if we want to actually delete the the access key in the object store.